code401challenges/src/main/java/code401challenges/linkedList/LinkedList.java

```diff
@@ -37,4 +39,29 @@ public String printValues() {
 
         return listValues + "{" + lastValue + "}" + "->" + "NULL";
     }
+
+    ////////////////////// Challenge 6 /////////////////////////////
+    public void append(int value) {
+        if (this.head == null) {
+            this.head = new Node(value);
+            this.head.next = null;
+        } else {
+            Node currentNode = head;
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = new Node(value, null);
+        }
+    }
+
+    public void insertBefore(int value, int newValue) {
+        Node currentNode = this.head;
```

Comment:
Check that list contains value before you run your while loop, otherwise you'll null pointer errors.